src/fondant/pipeline/lightweight_component.py

```diff
@@ -28,11 +32,42 @@ class PythonComponent(BaseComponent):
     def image(cls) -> Image:
         raise NotImplementedError
 
+    @classmethod
+    def consumes(cls) -> t.Optional[t.Dict[str, t.Any]]:
+        pass
+
+    @classmethod
+    def get_consumes_spec(
+        cls,
+        dataset_fields: t.Mapping[str, Field],
+    ):
+        consumes = cls.consumes()
+
+        if consumes is None:
+            # Get consumes spec from the dataset
+            spec_consumes = {k: v.type.to_dict() for k, v in dataset_fields.items()}
+
+            logger.warning(
+                "No consumes defined. Consumes will be inferred from the dataset."
+                "All field will be consumed which may lead to additional computation,"
+                "Consider defining consumes in the component.\n Consumes: %s",
+                spec_consumes,
+            )
+
+        elif consumes == {"additionalProperties": True}:
+            spec_consumes = consumes
+
+        else:
+            spec_consumes = {k: Type(v).to_dict() for k, v in consumes.items()}
```

Comment:
Shouldn't this be combined?

```suggestion
        else:
            spec_consumes = {
                k: (Type(v).to_dict() if k != "additionalProperties else v)
                for k, v in consumes.items()
            }
```